Initial content updates

Amruta-Ranade opened the pull request:
Partially addresses #3415 

Summary of changes:
- New include for UI cert content
- Updated the `Create Security Certificates` doc and `Use the Admin UI` doc with the new include.


To-Do:
- Add `include` to deployment and start-a-secure-cluster guides
- Test the procedure: Figure out how to configure the DNS names on one of the cloud platforms, use the DNS name to generate a certificate from Let's Encrypt, and then run through our documented steps.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/3959)
<!-- Reviewable:end -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d47a527c78ca69347b4d524cea3b2a6cd4aaab0c/

mberhault commented on v2.1/create-security-certificates.md:
```diff
@@ -9,7 +9,7 @@ toc: true
 </div>
 
-A secure CockroachDB cluster uses [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) for encrypted inter-node and client-node communication, which requires CA, node, and client certificates and keys. To create these certificates and keys, use the `cockroach cert` [commands](cockroach-commands.html) with the appropriate subcommands and flags, or use [`openssl` commands](https://wiki.ope ...
+A secure CockroachDB cluster uses [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) for encrypted inter-node and client-node communication, which requires Certificate Authority (CA), node, and client certificates and keys. To create these certificates and keys, use the `cockroach cert` [commands](cockroach-commands.html) with the appropriate subcommands and flags, or use [`openssl` co ...
```
This makes it sound like we need the CA key. We should clarify to make it clear that we only need the CA certificate, but we need the node and client certificates and keys.

mberhault commented on v2.1/create-security-certificates.md:
```diff
@@ -45,6 +49,8 @@ File name pattern | File usage
 `client.<user>.crt` | Client certificate for `<user>` (e.g., `client.root.crt` for user `root`)
 `client.<user>.key` | Key for the client certificate
 
+If you have a certificate issued by a public CA to securely access the Admin UI, you need to place the certificate in the directory specified by the `--certs-dir` flag.
```
Can we mention `ui.crt` and `ui.key` here, and make it clear they're optional?

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/b236d18e379a86a9f968811f8b69bb7434d8c4a9/

mberhault commented:
LGTM.

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/5a6ddadb3a692ccd029b40d0589f0a71c6b5a94f/

Amruta-Ranade commented:
@jseldess Added the include to the manual deployment guides.
I need help figuring out the updates to the orchestrated deployment guides (and also figure out if the UI cert applies to them).
And as for the `Start a local cluster` guide, I don't think the UI certs should be recommended for the local clusters (as I understand it, the users cannot get UI certs for localhost clusters).

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/5a6ddadb3a692ccd029b40d0589f0a71c6b5a94f/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/c7430e5f75679bb98d320b3b0a0d795370765c5a/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/37daab1387ef8e7761f043c000d34ec158a69de4/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e7b277e1cba9b8453768e6b7adde0cb2c59d7b60/

Amruta-Ranade commented:
@jseldess I have made the changes that we discussed in Wednesday's 1:1
- Removed the _include file
- Added the UI cert and key content as an example in the `cockroach cert` doc
- Referenced the example from other docs (Admin UI navigation doc and all deployment guides)

Let me know if it looks okay now.

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/6a9de8fbcf3f751e55c640d2dc8e7bc5ce9853a0/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/089216736180a9b823ef7c0de50a196f05b62c96/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/f484f85f57d477012e99e1cf9dc2d48b3fbbc48e/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3bf5dcfcaae75a1edd02a593404af3880bd31911/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/4f728c01d46cc16cd9f6618230fad2e17d21e10e/
